### ISSUE TITLE
Crash when importing a large OpenAPI definition

### ISSUE THREAD

**Issue by gshpychka** [a85/POSTMan-Chrome-Extension#13042]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

Similar behavior to #7844 when importing the OpenAPI spec for the Microsoft Graph API (~50MB).

When using the web app, the import modal simply never progresses past "Reading files..."

### Steps To Reproduce

1. Import new API
2. Provide the URL to the OpenAPI specification: https://graphexplorerapi.azurewebsites.net/openapi?operationIds=*&graphVersion=beta&openApiVersion=3&format=yaml
3. The client crashes

### Operating System

macOS

### Postman Version

11.6.1

### Postman Platform

Both

### User Account Type

Signed In User

**Comment by gshpychka** [comment]
Upon trying again today, the desktop client still "crashes" - the windows just becomes empty an non-responsive.

The Web client no longer hangs on "Reading files" - I get the usual prompt for the import options, but the import never finishes: 
<img width="367" alt="image" src="https://github.com/user-attachments/assets/569d4dcd-c432-4aa6-a734-8086c76c484a">

**Comment by KenzK** [comment]
I have also noticed this issue. when loading swagger.json file (~3.5MB) with almost 125000lines. Smaller version of the same swagger.json works fine. Another API tool when we used the same swagger.json worked fine. This could be a performance issue.

**Comment by gshpychka** [comment]
Not sure if this is relevant, but it seems that the web app is trying to communicate with the desktop client when importing and fails. I see the following in the browser console:

```
Referrer Policy: Ignoring the less restricted referrer policy “no-referrer-when-downgrade” for the cross-site request: http://localhost:10533/knockknock [import:15:19563](https://galactic-zodiac-883673.postman.co/workspace/Team-Workspace~803c60c0-4d0b-4555-8dab-5911552811b8/import)
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:10533/knockknock. (Reason: CORS request did not succeed). Status code: (null).
```

**Comment by gshpychka** [comment]
Not sure if this is relevant, but it seems that the web app is trying to communicate with the desktop client when importing and fails. I see the following in the browser console:

```
Referrer Policy: Ignoring the less restricted referrer policy “no-referrer-when-downgrade” for the cross-site request: http://localhost:10533/knockknock [import:15:19563](https://<our-instance>.postman.co/workspace/Team-Workspace~<id>/import)
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:10533/knockknock. (Reason: CORS request did not succeed). Status code: (null).
```

**Comment by VShingala** [comment]
@gshpychka Thanks for reporting the issue, we're looking into this and we'll share update here once we find something on this.

**Comment by gshpychka** [comment]
> The Web client no longer hangs on "Reading files" - I get the usual prompt for the import options, but the import never finishes: <img alt="image" width="367" src="https://private-user-images.githubusercontent.com/23005347/354128116-569d4dcd-c432-4aa6-a734-8086c76c484a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxOTMzODksIm5iZiI6MTcyMzE5MzA4OSwicGF0aCI6Ii8yMzAwNTM0Ny8zNTQxMjgxMTYtNTY5ZDRkY2QtYzQzMi00YWE2LWE3MzQtODA4NmM3NmM0ODRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA5VDA4NDQ0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiZWU5Nzg0NDdhZTRkN2M2NWY1ZDRkNjVlZTBmYzYxYmRjYTg1ZGRhODg0OTliZTBmZTc1YjhmMjdmYjRjOWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QaJFObWYNbEtjGB5NyYeLNf2WvvL5M0AP1bUVTFflpo">

I've since realized that it only hangs on `Reading files...` when importing from a URL. When importing from file, I see the behavior in the screenshot. In other words, nothing has changed, apologies for the confusion.

**Comment by gshpychka** [comment]
> @gshpychka Thanks for reporting the issue, we're looking into this and we'll share update here once we find something on this.

Have you been able to reproduce this at least?

**Comment by VShingala** [comment]
@gshpychka Yes, The definition that we're dealing with here is quite large (~46 MB). And the collection we generate out of it is also quite large (~480 MB) by using the [module directly](https://github.com/postmanlabs/openapi-to-postman?tab=readme-ov-file#-command-line-interface) and providing 16GB of heap space.

Due to such large sizes, we're running out of heap memory while doing this conversion from Postman App. Since, it's not a simple fix and requires us to take a deep look at the problem and solution we have ATM, it might take us good amount of time. We can't share the exact timeline for the fix. But it'd help us if you could answer following questions to help us for the fix.

1. What's the usecase specifically for converting this definition? Is it limited to certain endpoints from this definition?
2. If your use-case is not limited to certain endpoints, is it for documentation purposes?
3. Is this the only definition where you are facing the issue or have you face similar issue in past? If yes, could you share the definitions?

**Comment by gshpychka** [comment]
@VShingala thanks for the response. Yes, it is mainly for documentation, and also for an easy way to send requests (I have Authorization set up for Graph, now I need all the endpoints to be there). I do use a subset of endpoints, but postman does not allow me to pick a subset. That microsoft endpoint that produces the openapi spec does allow for glob patterns, but that's not really something that could produce a spec that I need.

I have not tried importing any other large definitions, so this is the first time I'm running into this issue.

**Comment by VShingala** [comment]
@gshpychka Thanks for providing your use case. I understand it and it makes sense.

The only issue seems to be sheer size. Although 50 MB size of OpenAPI may not be that high, the collection it generates is very large because data in collection is not reusable unlike OpenAPI definition where schemas can be just defined once and used in many places.

We're looking into more ways how we can make reduce the size of collection generated. But meanwhile, to unblock yourself for now I would suggest to only keep endpoints that you might be using and import the modified OpenAPI definition instead of importing entire definition.
